### PR TITLE
TASK: Add MathUtilityTypeSpecifyingExtension

### PR DESCRIPTION
**Pull request by sabbelasichon**
Resolves: #107

### CODE REVIEW

File: tests/Unit/Type/MathUtilityTypeSpecifyingExtension/data/MathUtilityTest.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+			return;
+		}
+
+		if (MathUtility::isIntegerInRange($c, 0, 200)) {
+			return;
+		}
+
+		assertType('int<0, 200>', $a);
+		assertType('*NEVER*', $b);
+		assertType('*NEVER*', $c);
```
Comment (sascha-egerer) [inline tests/Unit/Type/MathUtilityTypeSpecifyingExtension/data/MathUtilityTest.php:44]: I think this is not totally correct. The value will be casted to an integer by isIntegerInRange. The phpdoc says an integer must be given as value. So... should we keep it like that? I mean phpstan should detect the "wrong usage" anyways but someone could still ignore it an then this should not return never but also `int<0, 200>`

Comment (sabbelasichon) [reply tests/Unit/Type/MathUtilityTypeSpecifyingExtension/data/MathUtilityTest.php:44]: I think it is correct. No casting happens here: https://github.com/TYPO3/typo3/blob/main/typo3/sysext/core/Classes/Utility/MathUtility.php#L196

**Comment by sabbelasichon** [comment]
@sascha-egerer Would you mind switching to ECS and PSR-12? I really dislike the configuration of PHPCS

File: tests/Unit/Type/MathUtilityTypeSpecifyingExtension/data/MathUtilityTest.php (hunk 1 above)
Comment (sascha-egerer) [reply tests/Unit/Type/MathUtilityTypeSpecifyingExtension/data/MathUtilityTest.php:44]: Ah it's different in master than in v11. 
But it works for strings but not for float. See https://3v4l.org/k9GTl

**Comment by sascha-egerer** [comment]
> @sascha-egerer Would you mind switching to ECS and PSR-12? I really dislike the configuration of PHPCS

I feel you. But i would actually like to stay in sync with phpstan/phpstan-src. Opinions?

**Comment by sabbelasichon** [comment]
@sascha-egerer I have fixed the CS issues. But still think, that i would be more enjoyable to work with ECS.

File: tests/Unit/Type/MathUtilityTypeSpecifyingExtension/data/MathUtilityTest.php (hunk 1 above)
Comment (sabbelasichon) [reply tests/Unit/Type/MathUtilityTypeSpecifyingExtension/data/MathUtilityTest.php]: I have fixed it. Thanks for the tests

**Review by sascha-egerer — APPROVED** [review]

**Pull request merged**